src/test/regress/expected/pg16.out

```diff
@@ -1090,14 +1055,182 @@ RESET citus.enable_ddl_propagation;
+(1 row)
+
+DROP ROLE role6, role7;
+RESET citus.log_remote_commands;
+-- Grant role with admin option set to true and inherit option set to false
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role9;
+CREATE ROLE role10;
+GRANT role9 TO role10 WITH ADMIN OPTION, INHERIT FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role9' ORDER BY 1, 2;
+ role | member | admin_option
+---------------------------------------------------------------------
+ role9 | role10 | t
+(1 row)
+
+DROP ROLE role9, role10;
+RESET citus.log_remote_commands;
+-- Grant role with admin option set to false and inherit option set to true
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role11;
+CREATE ROLE role12;
+GRANT role11 TO role12 WITH INHERIT FALSE, ADMIN TRUE, SET FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
+  role  | member | admin_option | inherit_option | set_option
+---------------------------------------------------------------------
+ role11 | role12 | t            | f              | f
+(1 row)
+
+DROP ROLE role11, role12;
+RESET citus.log_remote_commands;
+-- Handles the inherit option
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role13;
+CREATE ROLE role14;
+GRANT role13 TO role14 WITH INHERIT FALSE;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option FROM pg_auth_members WHERE roleid::regrole::text = 'role13' ORDER BY 1, 2;
+  role  | member | admin_option | inherit_option
+---------------------------------------------------------------------
+ role13 | role14 | f            | f
+ (1 row)
+
+DROP ROLE role13, role14;
+RESET citus.log_remote_commands;
+-- Handles the set option
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role17;
+CREATE ROLE role18;
+GRANT role17 TO role18 WITH SET TRUE;
+SET ROLE role18;
+SELECT rolname, rolinherit FROM pg_roles WHERE rolname = 'role18';
+ rolname | rolinherit
+---------------------------------------------------------------------
+ role18  | t
+ (1 row)
+
+RESET ROLE;
+DROP ROLE role17, role18;
+RESET citus.log_remote_commands;
+-- Handles with default inherit false
+SET citus.enable_ddl_propagation TO on;
+CREATE ROLE role19;
+CREATE ROLE role20 NOINHERIT;
+GRANT role19 TO role20;
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option FROM pg_auth_members WHERE roleid::regrole::text = 'role19' ORDER BY 1, 2;
+  role  | member | admin_option
+---------------------------------------------------------------------
+ role19 | role20 | f
+(1 row)
+
+DROP ROLE role19, role20;
+RESET citus.log_remote_commands;
+-- show that creating role from worker node is allowed
+\c - - - :worker_1_port
+CREATE ROLE role_on_worker;
+DROP ROLE role_on_worker;
+\c - - - :master_port
+CREATE ROLE create_role1;
+CREATE ROLE create_role2;
```

Comment:
The "Test flakyness" schedules are failing right now because you need to drop the roles `create_role1` and `create_role2` at the end of this test.